lib/atlassian-jwt-authentication/filters.rb

```diff
@@ -136,7 +136,7 @@ def _verify_jwt(addon_key, consider_param: false, skip_qsh_verification: false,
 
       jwt_auth, account_id, context, qsh_verified = jwt_verification.verify
 
-      unless jwt_auth && (qsh_verified || skip_qsh_verification)
+      if jwt_auth == false || (!skip_qsh_verification && !qsh_verified)
```

Comment:
Are you looking for a boolean false or nil here? Maybe it will be better to use `jwt_auth.nil?`

Also this whole condition is very hard to read, maybe add some explanation or change variable names to be easier to understand?

Reply:
For boolean specifically.

Please explain what is hard to read in this condition. I'll do my best to improve readability, but honestly this is one of the areas in atlassian-jwt-authentication which I don't fully understand as well.